crates/store/src/state.rs

```diff
@@ -111,10 +111,20 @@ pub struct State {
 impl State {
     /// Loads the state from the `db`.
     #[instrument(target = COMPONENT, skip_all)]
-    pub async fn load(
-        mut db: Db,
-        block_store: Arc<BlockStore>,
-    ) -> Result<Self, StateInitializationError> {
+    pub async fn load(data_path: &Path) -> Result<Self, StateInitializationError> {
+        let data_directory = DataDirectory::load(data_path.to_path_buf())
+            .map_err(StateInitializationError::DataDirectoryLoadError)?;
+
+        let block_store = Arc::new(
+            BlockStore::load(data_directory.block_store_dir())
+                .map_err(StateInitializationError::BlockStoreLoadError)?,
+        );
+
+        let database_filepath = data_directory.database_path();
+        let mut db = Db::load(database_filepath.clone())
+            .await
+            .map_err(StateInitializationError::DatabaseLoadError)?;
+
         let nullifier_tree = load_nullifier_tree(&mut db).await?;
         let chain_mmr = load_mmr(&mut db).await?;
         let account_tree = load_accounts(&mut db).await?;
```

Comment:
Not from this PR, but I'd probably change the order to be:

1. Load MMR.
2. Load account tree.
3. Load nullifier tree.